### ISSUE TITLE
onSuggestionSelected callback removes focus from text field

### ISSUE THREAD

**Issue by fsobanski** [AbdulRahmanAlHamali/flutter_typeahead#98]
Hi,

when the `onSuggestionSelected` callback is invoked, it causes the text field to lose focus. Is it possible to keep the text field focused and the keyboard further open so that the user can continue editing the selected suggestion?

Output of flutter doctor:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.5.4-hotfix.1, on Linux, locale en_US.UTF-8)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Android Studio (version 3.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2018.1)
[✓] Connected device (1 available)

• No issues found!
```
Best,
Fabian

**Comment by KaYBlitZ** [comment]
Not currently. Not sure if you are using this widget for its intended purpose? The user should just be choosing the one suggestion that they are looking for, not editing the results that come up.

**Comment by fsobanski** [comment]
Thank you for your answer. In my use case, I have suggestions that are often very similar to the input that the user wants to type. I can't predict the user input exactly, all of the time.

**Comment by KaYBlitZ** [comment]
I can add a new property `keepSuggestionsOnSuggestionSelected` (quite a mouthful) to TypeAhead. The only problem I see with this is that you have to be very careful with users who are using physical keyboards. They won't be able to close the suggestions box unless they have a back button. iOS users will be unable to close the box at all....

**Comment by fsobanski** [comment]
It would be great if you would add that property. Otherwise, I would have probably done it at some point. It's just not my priority at the moment. 

For my mvp, I am only targeting android. I was also planning on dismissing the keyboard programmatically when the user taps somewhere outside of the keyboard,  the inputfield and the suggestion box. As far as I understand it, dismissing the keyboard should close the suggestion box as well.

**Comment by KaYBlitZ** [comment]
Yes, dismissing the keyboard will close the suggestions box. I can add a disclaimer or something in the README warning about problems with users using actual keyboards. Since they have physical keyboards, the soft keyboard will not show and cannot be dismissed.

**Comment by KaYBlitZ** [comment]
Fixed in 1.6.0
